### PR TITLE
Sh/doctor

### PR DESCRIPTION
**Pull request by shetzel**
### What does this PR do?
Exports `VersionDetail` needed by the Doctor class in plugin-info.
Adds 2 more properties to `VersionDetail`; the user's shell and the location of the sfdx-cli executable.

### What issues does this PR fix or reference?
@W-11489339@

### CODE REVIEW

**Review by peternhale — APPROVED** [review]

File: src/cli.ts [hunk 1]
```diff
@@ -118,29 +119,37 @@ function debugCliInfo(version: string, channel: string, env: Env, config: IConfi
...
+}
+
+// Function which returns Version object which includes cli version, plugin versions, OS and its version.
+function getVersionDetail(isVerbose: boolean, config: IConfig): VersionDetail {
+  const versions = config.userAgent.split(' ');
+  const cliVersion: string = versions[0];
+  const architecture: string = versions[1];
+  const nodeVersion: string = versions[2];
+  if (!isVerbose) return { cliVersion, architecture, nodeVersion };
+  const pluginVersion: string = exec('sfdx plugins --core', {
```
Comment (WillieRuemmele) [inline src/cli.ts:139]: can we get this info from the oclif config, or `plugin-plugins`? It might be faster than `exec`ing out for it

**Review by WillieRuemmele — APPROVED** [review]

File: src/cli.ts (hunk 1 above)
Comment (RodEsp) [reply src/cli.ts:139]: Yeah, we definitely should not be shelling out for this.
See [this](https://github.com/oclif/plugin-version/blob/65f6220479311714ed775cf6647cd2931f7b5bc7/src/commands/version.ts#L32-L36) as an example of how to get the plugins without running `sfdx plugins --core`.

**Review by RodEsp — COMMENTED** [review]
Maybe this is something that should exist in `@oclif/core`?
Or just be made available as an export from [@oclif/plugin-version](https://github.com/oclif/plugin-version/blob/main/src/commands/version.ts) where it already exists?

File: src/cli.ts (hunk 1 above)
Comment (shetzel) [reply src/cli.ts:139]: I'll look into it in the next PR.  I think you're right.

**Pull request closed without merging**